index.bs

```diff
@@ -138,6 +160,108 @@ call stack, at the time of the crash, to be included in a crash report.
+   document process crash.
+
+* The coordinator process maintains enough information about the state of each
+   [=Document=] that after a process crash, it can reconstruct the frame tree
+   that existed in each of that process' documents before the crash occurred.
+
+If these assumptions do not hold for a particular user agent, then this
+processing model may need to be modfied.
+
+These assumptions allow significant flexibility regarding the association of
+documents and processes. The following are all possible arrangments which this
+processing model is intended to support:
+
+* A process may control multiple [=Documents=]. These [=Documents=] may be part
+   of the same page, or may reside in different pages. They may be [=/top-level
+   traversables=] or [=child navigables=].
+* A process may control [=Documents=] which are cross-site or cross-origin to
+   each other.
+* A [=Document=] in a process which has crashed may have contained child
+   documents in processes which have not crashed
+* Similarly, a [=Document=] in a process which has crashed may be contained
+   within a parent document in a process which has not crashed.
+
+Registering Crash Reports {#registration}
+-------------------------
+
+The [=coordinator process=] has an <dfn>endpoint map</dfn>, which is a map of
+[=Documents=] -> [=endpoints=], initially empty.
+
+The [=coordinator process=] has a <dfn>process map</dfn>, which is a map of
+process -> [=lists=] of [=Documents=].
+
+Note: The maintenance of the [=process map=] is out of scope for this
+specification. It is intended that the user agent track, for every process it
+creates for controlling [=Documents=], a [=list=] of which [=Documents=] are
+controlled by that process.
+
+When a |document| [=initialize a global's endpoint list|initializes its endpoint
+list=], it should run the following step:
+
+1. If |document|'s [=endpoints=] contains `"default"`, then set
+    [=endpoint map=][|document|] to |document|'s [=endpoints=]["default"].
+
+When a document is destroyed, remove it from the map.
+
+Processing Crashes {#processing-crash}
+------------------
+
+<div class="algorithm" data-algorithm="process-crash">
+Given a crashed process |process|, with a string |reason|, this algorithm sends
+crash reports to the appropriate endpoints:
+
+1. Let |affectedDocuments| be [=process map=][|process|]
+1. For each |document| in |affectedDocuments|:
+  1. If [=endpoint map=][|document|] does not exist, continue.
+  1. Let |endpoint| be [=endpoint map=][|document|].
+  1. Let |parent| be |document|'s [=node navigable=]'s [=navigable/parent=]'s
+     [=active document=].
+  1. If |parent| is in |affectedDocuments|:
+    1. If [=endpoint map=][|parent|] = |endpoint|, continue.
+  1. Let |body| be a new {{CrashReportBody}}, initialized
+      as follows:
+
+        :   [=CrashReportBody/reason=]
+        ::  |reason|
+
+  1. Execute [=generate and queue a report=] with "crash", |endpoint|, and
```

Comment:
Is the Document Policy check and stack inclusion excluded on purpose? I can take a stab at adding in a follow-up PR if we want.